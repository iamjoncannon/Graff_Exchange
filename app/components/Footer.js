import React from "react";
import { logoUrl } from "./utils"

const front_end_url = "https://github.com/iamjoncannon/Graff_Exchange"
const back_end_url = "https://github.com/iamjoncannon/Graff_Exchange_API"

const Footer = (props) => {

    return (

        <footer>
                <a href={front_end_url} target="_blank"> 
                    <img src="https://jayceecodesphotos.s3.amazonaws.com/react.png" /> 
                </a>
                
                <a href={front_end_url} target="_blank"> 
                
                    <img src="https://jayceecodesphotos.s3.amazonaws.com/redux.png" />
                </a>

                <a href={front_end_url} target="_blank"> 

                    <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/96/Sass_Logo_Color.svg/2000px-Sass_Logo_Color.svg.png" style={{backgroundColor: "white", borderRadius: "10px"}}/>
                </a>

                <a href={back_end_url} target="_blank"> 

                    <img src={logoUrl} />
                </a>
                
                <a href={back_end_url} target="_blank"> 

                    <img src={"http://pluspng.com/img-png/nodejs-logo-png-nice-images-collection-node-js-desktop-wallpapers-370.png"} />
                </a>

                <a href={back_end_url} target="_blank"> 

                    <img src="https://cdn.iconscout.com/icon/free/png-512/redis-6-1175105.png" />
                </a>

                <a href={back_end_url} target="_blank"> 
                    <img src="https://jayceecodesphotos.s3.amazonaws.com/postgres.png" />
                </a>
                
                <a href={back_end_url} target="_blank"> 

                    <img src="https://jayceecodesphotos.s3.amazonaws.com/aws.png" style={{borderRadius: "10px"}}/>
                </a>

        </footer>
    )
}

export default Footer