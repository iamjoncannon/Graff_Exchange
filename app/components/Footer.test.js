import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './footer'

describe("Footer", ()=>{

    let component 

    beforeEach(()=>{

        component = renderer.create(<Footer 
                                        data={[1,2,3,5]}
                                        selectedDataNavItem={3}
                                    />)
    })

    it('1) renders successfully', ()=>{

        let tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
    })

    it("2) renders footer element", ()=>{

        const _footer = component.root.findByType("footer")
        
        expect(_footer).toBeTruthy()
    })

    it("3) renders eight anchor elements", ()=>{

        const anchors = component.root.findAllByType("a")

        expect(anchors.length).toEqual(8)
    })

    it("4) renders eight image elements", ()=>{

        const images = component.root.findAllByType("img")

        expect(images.length).toEqual(8)
    })
})