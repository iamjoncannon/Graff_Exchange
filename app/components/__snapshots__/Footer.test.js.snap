// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1) renders successfully 1`] = `
<footer>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange"
    target="_blank"
  >
    <img
      src="https://jayceecodesphotos.s3.amazonaws.com/react.png"
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange"
    target="_blank"
  >
    <img
      src="https://jayceecodesphotos.s3.amazonaws.com/redux.png"
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange"
    target="_blank"
  >
    <img
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/96/Sass_Logo_Color.svg/2000px-Sass_Logo_Color.svg.png"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": "10px",
        }
      }
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange_API"
    target="_blank"
  >
    <img
      src="../assets/graphql_logo.png"
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange_API"
    target="_blank"
  >
    <img
      src="http://pluspng.com/img-png/nodejs-logo-png-nice-images-collection-node-js-desktop-wallpapers-370.png"
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange_API"
    target="_blank"
  >
    <img
      src="https://cdn.iconscout.com/icon/free/png-512/redis-6-1175105.png"
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange_API"
    target="_blank"
  >
    <img
      src="https://jayceecodesphotos.s3.amazonaws.com/postgres.png"
    />
  </a>
  <a
    href="https://github.com/iamjoncannon/Graff_Exchange_API"
    target="_blank"
  >
    <img
      src="https://jayceecodesphotos.s3.amazonaws.com/aws.png"
      style={
        Object {
          "borderRadius": "10px",
        }
      }
    />
  </a>
</footer>
`;
